jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -307,6 +308,9 @@ protected synchronized void doStop()
         _servlets = shs;
         ServletMapping[] sms = (ServletMapping[])LazyList.toArray(servletMappings, ServletMapping.class);
         _servletMappings = sms;
+        
+        if (_contextHandler != null)
+            _contextHandler.contextDestroyed();
```

Comment:
We need a way to ensure that contextDestroyed is called by the ContextHandler if there is no ServletHandler present.

Reply:
In this PR or can this be done in a further PR?

Reply:
Let's get it done in this PR as it might need a little adjusting.
The issue being that it is asymmetric.... if there is a ServletHandler, then ContextHandler need not call contextDestroyed, but ContextHandler doesn't know about ServletHandlers.

So perhaps ContextHandler should have a marker interface called something like ContextDestroyer, that ServletHandler can implement.   It will have no methods, but ContextHandler can check if any of it's contained handlers implements ContextDestroyer, and if so then it doesn't call contextDestoyed.